tests/integration/delivery/delivery_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2021 VMware
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package delivery_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/cartographer/pkg/utils"
+)
+
+var _ = Describe("Deliveries", func() {
+	var (
+		ctx      context.Context
+		delivery *unstructured.Unstructured
+		cleanups []client.Object
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+	})
+
+	Describe("I can define a delivery with a resource", func() {
+		BeforeEach(func() {
+			deliveryYaml := utils.HereYaml(`
+				---
+				apiVersion: carto.run/v1alpha1
+				kind: ClusterDelivery
+				metadata:
+				  name: my-delivery
+				spec:
+			      resources:
+			        - name: my-first-resource
+					  templateRef:
+						kind: ClusterSourceTemplate
+						name: my-source-template
+			`)
+
+			delivery = &unstructured.Unstructured{}
+			err := yaml.Unmarshal([]byte(deliveryYaml), delivery)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			for _, obj := range cleanups {
+				_ = c.Delete(ctx, obj, &client.DeleteOptions{})
+			}
+		})
+
+		Context("the referenced resource exists", func() {
+			BeforeEach(func() {
+				clusterSourceTemplateYaml := utils.HereYaml(`
+                  ---
```

Comment:
same as the other comment - for things like this, I find using vim w/ `:set list` very useful (although replacing `\t` by anything visible would do the trick too I guess)